packages/teleport/src/Welcome/NewCredentials/NewMfaDevice.tsx

```diff
@@ -0,0 +1,265 @@
+import { ArrowBack } from 'design/Icon';
+import FieldInput from 'shared/components/FieldInput';
+import Validation, { Validator } from 'shared/components/Validation';
+import {
+  requiredField,
+  requiredToken,
+} from 'shared/components/Validation/rules';
+import createMfaOptions from 'shared/utils/createMfaOptions';
+import { SliderProps } from 'teleport/components/StepSlider';
+import { Props as CredentialsProps, LoginFlow } from './NewCredentials';
+import secKeyGraphic from './sec-key-with-bg.png';
+import { Auth2faType } from 'shared/services';
+
+export default function NewMfaDevice(props: Props) {
+  const {
+    resetToken,
+    submitAttempt,
+    clearSubmitAttempt,
+    auth2faType,
+    onSubmitWithWebauthn,
+    onSubmit,
+    password,
+    prev,
+    refCallback,
+  } = props;
+  const [otp, setOtp] = useState('');
+  const mfaOptions = useMemo(
+    () =>
+      createMfaOptions({
+        auth2faType: auth2faType,
+      }),
+    []
+  );
+  const [transitionPropertyName, setTransitionPropertyName] =
+    useState('height');
+  const [mfaType, setMfaType] = useState(mfaOptions[0]);
+  const [deviceName, setDeviceName] = useState(() =>
+    getDefaultDeviceName(mfaType.value)
+  );
+
+  useEffect(() => {
+    setTransitionPropertyName('');
+  }, []);
+
+  function onBtnClick(
+    e: React.MouseEvent<HTMLButtonElement>,
+    validator: Validator
+  ) {
+    e.preventDefault(); // prevent form submit default
+
+    if (!validator.validate()) {
+      return;
+    }
+
+    switch (mfaType?.value) {
+      case 'webauthn':
+        onSubmitWithWebauthn(password, deviceName);
+        break;
+      default:
+        onSubmit(password, otp, deviceName);
+    }
+  }
+
+  function onSetMfaOption(index: number, validator: Validator) {
+    setOtp('');
+    clearSubmitAttempt();
+    validator.reset();
+
+    const mfaOpt = mfaOptions[index];
+    setMfaType(mfaOpt);
+    setDeviceName(getDefaultDeviceName(mfaOpt.value));
+  }
+
+  const imgSrc =
+    mfaType?.value === 'otp'
+      ? `data:image/png;base64,${resetToken.qrCode}`
+      : secKeyGraphic;
+
+  return (
+    <Validation>
+      {({ validator }) => (
+        <Box p={5} ref={refCallback}>
+          <Flex mb={3} alignItems="center">
+            <ArrowBack
+              fontSize={30}
+              mr={3}
+              onClick={() => {
+                clearSubmitAttempt();
+                prev();
+              }}
+              style={{ cursor: 'pointer' }}
+            />
+            <Box>
+              <Text color="text.secondary">Step 2 of 2</Text>
+              <Text typography="h4" color="light" bold>
+                Set Two-Factor Device
+              </Text>
+            </Box>
+          </Flex>
+          {submitAttempt.status === 'failed' && (
+            <Danger children={submitAttempt.statusText} />
+          )}
+          <Text typography="subtitle1" color="light" caps mb={1}>
+            Two-Factor Method
+          </Text>
+          <Box>
```

Comment:
Spacing is a little off below the radio buttons: 
![image](https://user-images.githubusercontent.com/56373201/167910509-16e0fad5-4e38-4583-a3c1-7e566ed9edb7.png)


```suggestion
          <Box mb={1}>
```

Reply:
shifted things around a bit, but the change is here: https://github.com/gravitational/webapps/pull/818/files#diff-7a98121ef6a3ad155b6fb45127d9ba29b19bf15393be9bf92c555c0cf7712e33R126